source/server/overload_manager_impl.cc

```diff
@@ -44,12 +79,19 @@ class ThreadLocalOverloadStateImpl : public ThreadLocalOverloadState {
   const OverloadActionState& getState(const std::string& action) override {
     auto it = actions_.find(action);
     if (it == actions_.end()) {
-      it = actions_.insert(std::make_pair(action, OverloadActionState::Inactive)).first;
+      it = actions_.insert(std::make_pair(action, OverloadActionState::inactive())).first;
     }
     return it->second;
   }
 
-  void setState(const std::string& action, OverloadActionState state) { actions_[action] = state; }
+  void setState(const std::string& action, OverloadActionState state) {
+    auto it = actions_.find(action);
+    if (it == actions_.end()) {
+      actions_.emplace(action, state);
+    } else {
+      it->second = state;
+    }
```

Comment:
Would actions_.insert_or_assign(action, state); simplify the code above?

Reply:
Yep, done.